superset/charts/data/api.py

```diff
@@ -71,6 +76,7 @@ class ChartDataRestApi(ChartRestApi):
         action=lambda self, *args, **kwargs: f"{self.__class__.__name__}.data",
         log_to_statsd=False,
     )
+    @handle_api_exception
```

Comment:
Either @betodealmeida or @dpgaspar would probably be best to review this.

Reply:
Unsure about these, since the endpoints only handle 404 and 400 by themselves so every other possibility either succeeds or raises and exception (assuming `command.validate()` doesn't return flask responses).
So all exceptions are already handled by `@superset_app.errorhandler`

yet, `SupersetSecurityException`, `IntegrityError`, `DatabaseError`, `DataError` specially handled by `@handle_api_exception` seem to be missing from our flask errorhandlers.

Reply:
I'm now thinking about the tradeoffs between generic app-level handlers VS doing it through decorators. Maybe a better approach would be to factor out the error-handlers into a nice module, and make sure all cases are handled: API vs HTML and all potential errors and good catch-alls. @betodealmeida pointed out this solid SIP https://github.com/apache/superset/issues/9298 that @john-bodley wrote in 2020. Maybe now is the time to revive it and make it a reality.